### PR TITLE
Change LogModule to LogMonitoring

### PR DESCRIPTION
**Pull request by gkrenn**
## Description

The requirements for `LogModule` have changed: 

- `LogModule` should be called `LogMonitoring`
- `spec.logModule.enabled: true` ==> `spec.logMonitoring: {}`
- `spec.logModule.enabled: false` ==> `spec.logMonitoring: nil`

[Jira](https://dt-rnd.atlassian.net/browse/DAQ-1400)

## How can this be tested?

- Turn `LogMonitoring` on as described above => the DaemonSet should be created

### CODE REVIEW

File: pkg/api/v1beta3/dynakube/logmonitoring_props.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+		Spec: dk.Spec.LogMonitoring,
+	}
+	lm.SetName(dk.Name)
+
+	return lm
+}
+
+func (dk *DynaKube) LogMonitoringTemplates() logmonitoring.TemplateSpec {
+	return dk.Spec.Templates.LogMonitoring
+}
```
Comment (0sewa0) [inline pkg/api/v1beta3/dynakube/logmonitoring_props.go:18]: do it more like 

in `pkg/api/v1beta3/dynakube/logmonitoring/spec.go`
```
type LogMonitoring struct {
	*Spec
	*Templates

	name string
}
```

in `pkg/api/v1beta3/dynakube/logmonitoring_props.go`

```
func (dk *DynaKube) LogMonitoring() *logmonitoring.LogMonitoring {
	lm := &logmonitoring.LogMonitoring{
		Spec: dk.Spec.LogMonitoring,
		Templates: dk.Spec.Templates.LogMonitoring
	}
	lm.SetName(dk.Name)

	return lm
}
```

So you can do the following `dk.LogMonitoring().DNSPolicy` and such.
IMO, it is much nicer than `dk. LogMonitoringTemplates().DNSPolicy`
Or you can also do `dk.LogMonitoring().Templates.DNSPolicy` if there is naming conflict

**Review by 0sewa0 — CHANGES_REQUESTED** [review]
Looks great, left a small suggestion

File: pkg/api/v1beta3/dynakube/logmonitoring_props.go (hunk 1 above)
Comment (0sewa0) [reply pkg/api/v1beta3/dynakube/logmonitoring_props.go:18]: if there ever be a `logmonitoring.Status` you would also put that into the `type LogMonitoring struct`, so we truly handle it as their own CR

Comment (gkrenn) [reply pkg/api/v1beta3/dynakube/logmonitoring_props.go]: 👍  - Done

File: pkg/api/v1beta3/dynakube/logmonitoring/props.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const (
+	logModuleDaemonSetSuffix = "-logmodule"
+)
+
+func (logMonitoring *Spec) SetName(name string) {
+	logMonitoring.name = name
+}
+
+func (logMonitoring *Spec) Needed() bool {
```
Comment (waodim) [inline pkg/api/v1beta3/dynakube/logmonitoring/props.go]:
```suggestion
func (logMonitoring *Spec) IsNeeded() bool {
```

Comment (waodim) [reply pkg/api/v1beta3/dynakube/logmonitoring/props.go]: Don't have the strongest opinion for this one - I just think it fits the rest of our codebase better. You could also go with `Enabled()` but this is just my opinion, maybe you and/or the others are fine with `Needed()`

**Review by waodim — COMMENTED** [review]
only one nitpick, otherwise lgtm

**Review by waodim — APPROVED** [review]
code lgtm - logmonitoring ds starts as expected.

**Review by 0sewa0 — APPROVED** [review]
Tested it works, small leftover https://github.com/Dynatrace/dynatrace-operator/blob/caf163e3ac7ad427bac6191c2d019a7ff657cddc/pkg/controllers/dynakube/logmonitoring/daemonset/reconciler.go#L60 (due to change that happened after this PR was published)

**Pull request merged**